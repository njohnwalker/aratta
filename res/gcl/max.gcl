@req(true)
if x >= y -> m := x 
[] y >= x -> m := y
fi
@ens(m >= x & m >= y)