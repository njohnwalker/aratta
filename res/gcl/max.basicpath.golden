Path 1
  [ Branch 1
      ; Assume x >= y
      ; Substitute
          @m -> x
      { HOLE }
    Branch 2
      ; Assume y >= x
      ; Substitute
          @m -> y
      { HOLE }
  ]
  ; Substitute
      @n -> m
  { Satisfies n >= x & n >= y }